### PR TITLE
chore: event timeline

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-2657/implement-a-first-iteration-of-an-horizontal-event-timeline

This implements the very first iteration of our event timeline.

This is behind a feature flag, which when enabled shows the new timeline at the top of our event log page.

It is missing some features, like:
 - Placement: It should show up as an option in the header, not in the event log;
 - Tooltip: It should show proper tooltips for all the events that we're displaying;
 - Grouping: It should group together events that occurred in a short span of time;
 - Signals: It should show signals along with the events;

Here's how it currently looks like, with some example events, in order from left to right:
 - A flag was disabled more than 30 min ago;
 - A flag was then enabled;
 - A segment was updated (didn't have an icon for segments, so I picked one);
 - A strategy was updated;
 - A flag was created;

image
image

(Time passed since I took the first screenshot, so you can see the events "moved" to the left slightly in the dark theme screenshot)

I have some concerns about the low contrast of `neutral` variant events, especially in dark mode. Maybe we should consider using `error` instead, for red? Or maybe add a border to our event circles?

I specifically changed my environment to be "development" for the screenshots. The default selection is the first enabled environment that is `type=production`, which in my case is "production".

Here are our filters:
 - Time Span
image
 - Environment
image

Here are a few more screenshots, with the different time spans (zooming out, since we're increasing the time span):

image
image
image
image
image
image

Again, when zooming out, some events should be grouped together, but that's a task for later.

### CODE REVIEW

File: frontend/src/component/events/EventTimeline/EventTimeline.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    'feature-created',
+    'feature-updated',
+    'feature-variants-updated',
+    'feature-archived',
+    'feature-revived',
+    'feature-strategy-update',
+    'feature-strategy-add',
+    'feature-strategy-remove',
+    'feature-environment-enabled',
+    'feature-environment-disabled',
```
Comment (nunogois) [inline frontend/src/component/events/EventTimeline/EventTimeline.tsx:80]: These are the events we're currently considering for the timeline.

File: frontend/src/component/events/EventTimeline/EventTimeline.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    );
+
+    const filteredEvents = events.filter(
+        (event) =>
+            new Date(event.createdAt).getTime() >= startDate.getTime() &&
+            new Date(event.createdAt).getTime() <= endDate.getTime() &&
+            RELEVANT_EVENT_TYPES.includes(event.type) &&
+            (!event.environment ||
+                !environment ||
+                event.environment === environment.name),
```
Comment (nunogois) [inline frontend/src/component/events/EventTimeline/EventTimeline.tsx:110]: Here I decided that events without a specific environment should always show regardless of our environment selection.

File: frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEvent.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    if (type.startsWith('feature-')) {
+        return <FlagOutlinedIcon />;
+    }
+    if (type.startsWith('segment-')) {
+        return <GroupsIcon />;
+    }
+
+    return <QuestionMarkIcon />;
+};
```
Comment (nunogois) [inline frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEvent.tsx:67]: This is where we pick which icon will show for each event type.

File: frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEvent.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+const customEventVariants: Partial<
+    Record<EventSchemaType, CustomEventVariant>
+> = {
+    'feature-environment-enabled': 'success',
+    'feature-environment-disabled': 'neutral',
+    'feature-archived': 'neutral',
+};
```
Comment (nunogois) [inline frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEvent.tsx:75]: And this is where we specify special variants, like "green" or "grey" icons. Any other event types default to "secondary" which is our "light purple".

File: frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEventTooltip/EventTimelineEventTooltip.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    return (
+        <div>
+            <div>{eventDateTime}</div>
+            <div>{event.createdBy}</div>
+            <div>{event.type}</div>
+            <div>{event.featureName}</div>
+            <div>{event.environment}</div>
+        </div>
+    );
+};
```
Comment (nunogois) [inline frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEventTooltip/EventTimelineEventTooltip.tsx:50]: This needs to be properly implemented. Will be addressed at a later point.

File: frontend/src/component/events/EventTimeline/EventTimelineHeader/EventTimelineHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        () => environments.filter(({ enabled }) => enabled),
+        [environments],
+    );
+
+    useEffect(() => {
+        if (activeEnvironments.length > 0) {
+            const defaultEnvironment =
+                activeEnvironments.find(({ type }) => type === 'production') ||
+                activeEnvironments[0];
+            setEnvironment(defaultEnvironment);
```
Comment (nunogois) [inline frontend/src/component/events/EventTimeline/EventTimelineHeader/EventTimelineHeader.tsx:108]: Here is the logic for the default environment selection, as mentioned in the PR description.

File: frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEventTooltip/EventTimelineEventTooltip.tsx (hunk 5 above)
Comment (gastonfournier) [reply frontend/src/component/events/EventTimeline/EventTimelineEvent/EventTimelineEventTooltip/EventTimelineEventTooltip.tsx:50]: Cool, this looks like a rendering strategy. Agree to do it later!

**Review by gastonfournier — APPROVED** [review]
Very good and paying attention to details!

**Pull request merged**